#!/usr/bin/env python

PACKAGE="pcl_assembler"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("debug", bool_t,   0, "debug: ",   False)
gen.add("seg_radius",        double_t, 0, "segradius: ",          0.7,     0.0, 10.0)
gen.add("point_neighborhood_check", bool_t,   0, "Check the neighborhood to choose pole point: ",   True)
gen.add("r",        double_t, 0, "r around the point: ",          0.1,     0.0, 10.0)
gen.add("K",        int_t, 0, "K neighbors: ",          100,     1, 1000)
gen.add("min_near_points",        int_t, 0, "min near points: ",          5,     1, 200)
gen.add("use_LR_model", bool_t,   0, "Select the pole on the row model: ",   False)
gen.add("use_RANSAC_model", bool_t,   0, "Select the pole on the row model: ",   False)
gen.add("threshold",        double_t, 0, "threshold for RANSAC: ",          0.5,     0.0, 100.0)
gen.add("iterations",        int_t, 0, "RANSAC iterations: ",          100,     1, 10000)

exit(gen.generate(PACKAGE, "pcl_euclidean_filter", "ClusteringEuclideanParam"))
